examples/apple/coreml/scripts/inspector_utils.py

```diff
@@ -0,0 +1,429 @@
+
+
+def core_aten_to_edge(
+    core_aten_program: ExportedProgram,
+    edge_compile_config: exir.EdgeCompileConfig,
+) -> EdgeProgramManager:
+    edge_manager = to_edge(
+        programs=core_aten_program,
+        compile_config=edge_compile_config,
+    )
+    return edge_manager
+
+
+def module_to_edge(
+    module: torch.nn.Module,
+    example_inputs: Tuple[Value, ...],
+    edge_compile_config: exir.EdgeCompileConfig = _EDGE_COMPILE_CONFIG,
+) -> EdgeProgramManager:
+    module.eval()
+    core_aten_program = to_core_aten(
+        module=module,
+        example_inputs=example_inputs,
+    )
+    return core_aten_to_edge(
+        core_aten_program=core_aten_program,
+        edge_compile_config=edge_compile_config,
+    )
+
+
+def lower_and_export_edge_to_coreml(
+    edge_program: EdgeProgramManager,
+    compile_specs: List[CompileSpec],
+    config: ExecutorchBackendConfig,
+    skip_ops_for_coreml_delegation: Optional[List[str]] = None,
+) -> ExirExportedProgram:
+    partitioner = CoreMLPartitioner(
+        skip_ops_for_coreml_delegation=skip_ops_for_coreml_delegation,
+        compile_specs=compile_specs,
+    )
+    delegated_program_manager = edge_program.to_backend(
+        partitioner,
+    )
+    executorch_program = delegated_program_manager.to_executorch(
+        config=config,
+    )
+    return executorch_program
+
+
+def write_to_file(buffer: bytes, file_path: Path):
+    with open(file_path.resolve(), "wb") as file:
+        file.write(buffer)
+
+
+def generate_bundled_program(
+    executorch_program: ExecutorchProgramManager,
+    example_inputs: Tuple[Value, ...],
+    method_name: str,
+    bundled_program_path: Path,
+):
+    method_test_suites = [
+        MethodTestSuite(
+            method_name=method_name,
+            test_cases=[MethodTestCase(inputs=example_inputs)],
+        )
+    ]
+
+    bundled_program = BundledProgram(executorch_program, method_test_suites)
+    bundled_program_buffer = serialize_from_bundled_program_to_flatbuffer(
+        bundled_program
+    )
+
+    write_to_file(buffer=bundled_program_buffer, file_path=bundled_program_path)
+
+
+def generate_etdump_with_intermediate_values(
+    root_dir_path: Path,
+    bundled_program_path: Path,
+    et_dump_path: Path,
+    debug_buffer_path: Path,
+    debug_buffer_size: int,
+):
+    sdk_executable_path = (
+        root_dir_path / "cmake-out" / "examples" / "sdk" / "sdk_example_runner"
+    )
+    if not sdk_executable_path.is_file():
+        raise FileNotFoundError(
+            errno.ENOENT, os.strerror(errno.ENOENT), str(sdk_executable_path.resolve())
+        )
+
+    sdk_runner_command: str = f"""
+    {sdk_executable_path.resolve()} -dump_intermediate_outputs\
+    -bundled_program_path {bundled_program_path.resolve()}\
+    -etdump_path {et_dump_path.resolve()}\
+    -debug_output_path {debug_buffer_path.resolve()}\
+    -debug_buffer_size {debug_buffer_size}"""
+    run_command: str = f"{sdk_runner_command}"
```

Comment:
why the f-string? sdk_runner_command is already a string, couldn't this just read `run_command = sdk_runner_command`?

Reply:
Not needed, will remove it.